### PR TITLE
Fix/load more sort

### PR DESCRIPTION
**Pull request by Andries-Smit**

### CODE REVIEW

File: src/DropDownSort/components/DropDownSortContainer.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export interface FormState {
 
 export default class DropDownSortContainer extends Component<ContainerProps, ContainerState> {
     private dataSourceHelper?: DataSourceHelper;
-    private widgetDom: HTMLElement | null = null;
+    private widgetDom: Element | null = null;
```
Comment (IIsaku) [inline src/DropDownSort/components/DropDownSortContainer.ts]: I think the function "connectToListView" should also be changed over here.

File: src/HeaderSort/components/HeaderSortContainer.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ interface FormState {
 
 export default class HeaderSortContainer extends Component<ContainerProps, ContainerState> {
     private dataSourceHelper?: DataSourceHelper;
-    private widgetDom: HTMLElement | null = null;
+    private widgetDom: Element | null = null;
```
Comment (IIsaku) [inline src/HeaderSort/components/HeaderSortContainer.ts]: I think the function "connectToListView" should also be changed over here.

File: src/Pagination/components/PaginationContainer.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ interface PaginationPageState {
 }
 
 class PaginationContainer extends Component<ModelerProps, PaginationContainerState> {
-    private widgetDom: HTMLElement | null = null;
+    private widgetDom: Element | null = null;
```
Comment (IIsaku) [inline src/Pagination/components/PaginationContainer.ts]: Maybe the same change is necessary in "connectToListView" function.

File: tests/e2e/TextSearch/TextBoxSearch.spec.ts [hunk 4]
```diff
@@ -1,40 +1,41 @@
...
         page.searchInput.click();
         page.searchInput.setValue(testValue);
-
-        const listviewItems: Element[] = page.listViewList.value;
-        expect(listviewItems.length).toBe(1);
+        browser.pause(1000);
+        expect(page.listViewListItems.length).toBe(1);
     });
 
     it("when query is entered in the search input and clear button clicked the list view should be filtered with new query", () => {
```
Comment (IIsaku) [inline tests/e2e/TextSearch/TextBoxSearch.spec.ts]: It's good to check whether the query send after using the clear button is the correct one by verifying that the total list items at the beginning of the test is also 5.

File: src/Shared/DataSourceHelper/DataSourceHelper.ts [hunk 5]
```diff
@@ -194,7 +194,15 @@ export class DataSourceHelper {
                 this.widget._datasource._sorting = sorting;
             }
             logger.debug("DataSourceHelper .set sort and constraint");
-
+            if (!this.widget.__lvcPagingEnabled) {
+                const offset = this.widget._datasource.getOffset();
```
[the remark this answers is not in the document]
Comment (Andries-Smit) [reply src/Shared/DataSourceHelper/DataSourceHelper.ts]: have to set the paging to include till end, to force refresh of all items, when sorting is changed.
Just added, after update to restore the paging

File: src/DropDownSort/components/DropDownSortContainer.ts (hunk 1 above)
Comment (Andries-Smit) [reply src/DropDownSort/components/DropDownSortContainer.ts]: Not sure what you try to say? Are you referring to the error state fix? as this widget does not depend on context validation. it is not needed

File: tests/e2e/TextSearch/TextBoxSearch.spec.ts (hunk 4 above)
Comment (Andries-Smit) [reply tests/e2e/TextSearch/TextBoxSearch.spec.ts]: That is done on line 27, `searchButton` should be renamed to clearButton

File: src/Pagination/components/PaginationContainer.ts (hunk 3 above)
Comment (Andries-Smit) [reply src/Pagination/components/PaginationContainer.ts]: Does not depend on context, validation is based on props not on state

File: tests/e2e/TextSearch/TextBoxSearch.spec.ts (hunk 4 above)
Comment (Andries-Smit) [reply tests/e2e/TextSearch/TextBoxSearch.spec.ts]: done

File: src/HeaderSort/components/HeaderSortContainer.ts (hunk 2 above)
Comment (Andries-Smit) [reply src/HeaderSort/components/HeaderSortContainer.ts]: No validation based on stated

**Pull request merged**